### PR TITLE
Fix high cpu load due to log capture thread not release

### PR DESCRIPTION
**Pull request by ulysses-you**
We will create a new log capture thread when open a new session, but we forgot release the thread after session opened success or failure.

Note that, we don't need to wait sub process finished because the log capture lifecycle is `SESSION_CREATE`  -> `SESSION_OPEN`. After `SESSION_OPEN` the log is controled by session.

### CODE REVIEW

**Comment by ulysses-you** [comment]
cc @yaooqinn

File: kyuubi-main/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala [hunk 1]
```diff
@@ -106,28 +106,35 @@ class KyuubiSessionImpl(
...
-            throw KyuubiSQLException(s"Timed out($timeout ms) to launched Spark with $builder",
-              builder.getError)
-          }
-          sh = getServerHost
+          val Some((host, port)) = sh
+          openSession(host, port)
+        } finally {
+          // we must close the process builder whether session open is success or failure since
+          // we have a log capture thread in process builder.
+          builder.close()
```
Comment (yaooqinn) [inline kyuubi-main/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala:134]: Shall we give it a more specific name here like stopLoggingThread

**Review by yaooqinn — APPROVED** [review]

File: kyuubi-main/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala (hunk 1 above)
Comment (ulysses-you) [reply kyuubi-main/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala:134]: I think user(like `KyuubiSessionImpl`) does not need to care about whether `ProcessBuilder` use a log thread. `close` is a more common method that can remind user execute it.

**Comment by yaooqinn** [comment]
thanks! merge to master for 1.1 and branch 1.0 for 1.0.2

**Pull request closed without merging**